packages/monaco/src/browser/monaco-quick-input-service.ts

```diff
@@ -227,15 +227,23 @@ export class MonacoQuickInputService implements QuickInputService {
     }
 
     async pick<T extends QuickPickItem, O extends PickOptions<T> = PickOptions<T>>(
-        picks: Promise<QuickPickInput<T>[]> | QuickPickInput<T>[], options: O = <O>{}, token?: monaco.CancellationToken
+        picks: Promise<QuickPickInput<T>[]> | QuickPickInput<T>[], options?: O, token?: monaco.CancellationToken
     ): Promise<(O extends { canPickMany: true; } ? T[] : T) | undefined> {
+        type M = T & { buttons?: NormalizedQuickInputButton[] };
+        type R = (O extends { canPickMany: true; } ? T[] : T);
         const monacoPicks = (await picks).map(pick => {
-            if (pick.type === 'separator') {
-                return pick;
+            if (pick.type !== 'separator') {
+                pick.buttons &&= pick.buttons.map(QuickInputButton.normalize);
             }
-            return this.convertItems(pick);
+            return pick as M;
         });
-        return this.monacoService.pick(monacoPicks, options as any, token) as any; /* eslint-disable-line @typescript-eslint/no-explicit-any */
+        const monacoOptions = options as IPickOptions<M>;
+        const picked = await this.monacoService.pick(monacoPicks, monacoOptions, token);
+        if (!picked) { return picked; }
+        if (options?.canPickMany) {
+            return (Array.isArray(picked) ? picked : [picked]) as R;
+        }
+        return Array.isArray(picked) ? picked[0] : picked;
```

Comment:
This one I'm less sure if worth or not but I feel like there's a couple cases that shouldn't happen, though we can keep handling them like you did:

```suggestion
        if (options?.canPickMany) {
            if (!Array.isArray(picked)) {
                console.error('expected the returned picks to be an array');
                return [picked] as R;
            }
            return picked;
        }
        if (Array.isArray(picked)) {
            console.error('expected the returned pick to not be an array');
            return picked[0]; // if picked.length === 0 this will return undefined...
        }
        return picked as R;
```